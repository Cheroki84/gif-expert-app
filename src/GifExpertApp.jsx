import { useState } from "react";
import AddCategory from "./components/AddCategory";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball', 'One Piece', 'Pokemon']);
    
    // const handleAdd = () => {
    //     // setCategories([...categories, 'Naruto']);
    //     setCategories(cats => [...cats, 'Naruto']);
    // }

    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories} />
        <hr/>


        <ol>
            {
                categories.map(category => {
                    return <li key={category} >{category}</li>
                })
            }
        </ol>

        </>
    )
}

export default GifExpertApp;