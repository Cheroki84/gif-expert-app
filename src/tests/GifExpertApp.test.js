import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas sobre el componente GifExpertApp', () => {
    
    test('El componente debe mostrarse correctamente (snapshot)', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de categorÃ­as', () => {
        
        const categories = ['Dragon Ball', 'Pokemon']

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });

});
