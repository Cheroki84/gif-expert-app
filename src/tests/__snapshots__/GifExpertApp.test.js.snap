// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifExpertApp Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre el componente GifExpertApp El componente debe mostrarse correctamente (snapshot) 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
